10_rest/mohitnag/cmd/puppy-server/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy"
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/puppy/store"
+	"github.com/anz-bank/go-course/10_rest/mohitnag/pkg/rest"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app   = kingpin.New("puppyStore", "Puppy Store")
+	args  = os.Args[1:]
+	srvCh = make(chan *http.Server)
+)
+
+func main() {
+	var storeType string
+	fileName := app.Flag("data", "file path").Short('d').ExistingFile()
+	port := app.Flag("port", "port number").Short('p').Default("8686").String()
+	app.Flag("store", "store type").Short('s').Default("sync").EnumVar(&storeType, "map", "sync")
+	kingpin.MustParse(app.Parse(args))
+
+	s, err := initialisePuppyStoreWithFile(storeType, *fileName)
+	if err != nil {
+		panic(err)
+	}
+	handler := rest.NewRestHandler(s)
+	srv := &http.Server{
+		Addr:    ":" + (*port),
+		Handler: handler,
+	}
+	srvCh <- srv
+	if err := srv.ListenAndServe(); err != nil {
```

Comment:
i was not able to use `log.Fatal` as it shut down the server in case on any issue. Instead I have used `log.Panic(srv.ListenServe())`. Please have a look.